hypertrace-graphql-gateway-service-metric-utils/src/main/java/org/hypertrace/graphql/utils/metrics/gateway/BaselineMetricSeriesMapConverter.java

```diff
@@ -0,0 +1,61 @@
+package org.hypertrace.graphql.utils.metrics.gateway;
+
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.core.Single;
+import org.hypertrace.core.graphql.common.utils.BiConverter;
+import org.hypertrace.core.graphql.common.utils.CollectorUtils;
+import org.hypertrace.gateway.service.v1.baseline.BaselineMetricSeries;
+import org.hypertrace.graphql.metric.request.MetricSeriesRequest;
+import org.hypertrace.graphql.metric.schema.BaselineMetricInterval;
+
+import javax.inject.Inject;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+
+import static org.hypertrace.core.graphql.common.utils.CollectorUtils.immutableMapEntryCollector;
+
+class BaselineMetricSeriesMapConverter
+    implements BiConverter<
+        List<MetricSeriesRequest>,
+        Map<String, BaselineMetricSeries>,
+        Map<Duration, List<BaselineMetricInterval>>> {
+  private final BaselineIntervalListContainer intervalListConverter;
+
+  @Inject
+  BaselineMetricSeriesMapConverter(BaselineIntervalListContainer intervalListConverter) {
+    this.intervalListConverter = intervalListConverter;
+  }
+
+  @Override
+  public Single<Map<Duration, List<BaselineMetricInterval>>> convert(
+      List<MetricSeriesRequest> metricSeriesRequests,
+      Map<String, BaselineMetricSeries> baselineMetricSeriesMap) {
+    return Observable.fromIterable(metricSeriesRequests)
```

Comment:
Same as intervallist, this looks like copy paste that could be shared.

Reply:
As i said earlier will do refactoring in next iteration since it already became big patch.